### PR TITLE
Enemy spawning

### PR DESCRIPTION
**Pull request by pkasinski**
Enemies now spawn continuously, so that there is always a certain number of enemies in the world.

### CODE REVIEW

File: core/src/model/Model.java [hunk 1]
```diff
@@ -5,25 +5,38 @@
...
+    private final IMapLoader mapLoader;
     private IPlayerCharacter player;
-    private List<Enemy> enemyList = new ArrayList<>();
-    private List<Entity> entityList = new ArrayList<>();
-    private RewardSystem rewardSystem = new RewardSystem();
-    private World<IEntity> world = new World<>();
+    private final List<Enemy> enemyList = new ArrayList<>();
+    private final List<Entity> entityList = new ArrayList<>();
+    private final List<Point> spawnPoints;
+    private static final int MAX_ENEMIES = 8;
```
Comment (rojtekin) [inline core/src/model/Model.java:23]: This shouldnt be constant?

**Review by rojtekin — APPROVED** [review]
should change max amount of enemies to not a constant otherwise LGTM

File: core/src/model/Model.java [hunk 2]
```diff
@@ -60,12 +73,36 @@ public List<Enemy> getEnemyList() {
...
+            if (spawnPointsIndex < spawnPoints.size() - 1) { spawnPointsIndex++; }
+            else { spawnPointsIndex = 0; }
+        }
+    }
+
+    /**
+     * Adds a random enemy at the specified spawn point.
+     * @param spawnPoint the spawn point where a random enemy should spawn
+     */
+    private void spawnRandomEnemy(Point spawnPoint) {
```
Comment (Suedie) [inline core/src/model/Model.java:97]: This could be moved out into its own factory

File: core/src/model/Model.java [hunk 3]
```diff
@@ -60,12 +73,36 @@ public List<Enemy> getEnemyList() {
...
+        if (Math.random() < 0.75) {
+            addEnemy(new Cyclops(spawnPoint.x, spawnPoint.y,1,1,1, getWorld()));
+        }
+        else {
+            addEnemy(new Mouse(spawnPoint.x, spawnPoint.y,2,1,1, getWorld()));
+        }
+    }
```
Comment (Suedie) [inline core/src/model/Model.java:104]: This could be moved out into its own factory

File: core/src/model/ModelFactory.java [hunk 4]
```diff
@@ -1,15 +1,26 @@
...
+
+        int xLeft = 100;
+        int xCenter = mapLoader.getMapUnitWidth() / 2;
+        int xRight = mapLoader.getMapUnitWidth() - 100;
+        int yBottom = 100;
+        int yCenter = mapLoader.getMapUnitHeight() / 2;
+        int yTop = mapLoader.getMapUnitHeight() - 100;
+        List<Point> spawnPoints = Arrays.asList(new Point(xLeft, yTop), new Point(xCenter, yTop), new Point(xRight, yTop),
+                                                new Point(xLeft, yCenter), new Point(xRight, yCenter),
+                                                new Point(xLeft, yBottom), new Point(xCenter, yBottom), new Point(xRight, yBottom));
```
Comment (Suedie) [inline core/src/model/ModelFactory.java:20]: Won't the distance between spawnpoints be too great when we implement our larger map?

**Review by Suedie — APPROVED** [review]
lgtm, left some comments

File: core/src/model/ModelFactory.java (hunk 4 above)
Comment (pkasinski) [reply core/src/model/ModelFactory.java:20]: This may have to change if a larger map is implemented.

**Pull request merged**